lib/reek/source/source_path.rb

```diff
@@ -0,0 +1,89 @@
+require 'private_attr/everywhere'
+require 'find'
+require 'pathname'
+require 'forwardable'
+
+module Reek
+  module Source
+    #
+    # Source file path in a filesystem.
+    #
+    class SourcePath
+      extend Forwardable
+      include Enumerable
+
+      def_delegators :pathname, :read, :to_s
+      def_delegator :traverse_children, :each
+
+      # Initialize with the pathname and configuration.
+      #
+      # pathname - a path as Pathname
+      def initialize(pathname, configuration: Configuration::AppConfiguration.default)
+        @configuration = configuration
+        @pathname = pathname.cleanpath
+        ensure_file
+      end
+
+      # Checks is path is relevant
+      #
+      # @return Boolean
+      def relevant?
```

Comment:
Can you give this a more specific name? It is not immediately clear why any path that is not ignored would not be relevant.